packages/shared/src/contexts/components/InjectedDialog.tsx

```diff
@@ -138,14 +138,36 @@ export function InjectedDialog(props: InjectedDialogProps) {
     const actions = CopyElementWithNewProps(children, DialogActions, { root: dialogActions })
     const content = CopyElementWithNewProps(children, DialogContent, { root: dialogContent })
     const { extraProps, shouldReplaceExitWithBack, IncreaseStack } = useDialogStackActor(open)
+    const [recoverOverflow, setRecoverOverflow] = useState(false)
 
     const closeBothCompositionDialog = useCallback(() => {
         if (isOpenFromApplicationBoard) {
             CrossIsolationMessages.events.compositionDialogEvent.sendToLocal({ open: false, reason: 'timeline' })
         }
-
+        if (recoverOverflow) {
+            const html = document.documentElement
+            html.style.overflow = 'auto scroll'
+            html.style.removeProperty('margin-right')
+            setRecoverOverflow(false)
+        }
         onClose?.()
-    }, [isOpenFromApplicationBoard, onClose])
+    }, [isOpenFromApplicationBoard, onClose, recoverOverflow])
+
+    useEffect(() => {
+        const html = document.documentElement
+        if (open && html.style.overflow !== 'hidden') {
+            html.style.overflow = 'hidden'
+            html.style.marginRight = '17px'
```

Comment:
This does not look right. Do you mean to use `scrollLock` from mui `<Modal />` component?

https://mui.com/material-ui/api/modal/

Reply:
modal cannot replace dialog

Reply:
what do you mean? dialog is based on modal.